src/plugins/conductor/TimeConductorService.js

```diff
@@ -0,0 +1,170 @@
+ * Open MCT Web, Copyright (c) 2014-2015, United States Government
+ * as represented by the Administrator of the National Aeronautics and Space
+ * Administration. All rights reserved.
+ *
+ * Open MCT Web is licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * Open MCT Web includes source code licensed under additional open source
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+
+define(
+    [
+        'EventEmitter'
+    ],
+    function (EventEmitter) {
+
+        /**
+         * A class representing the state of the time conductor view. This
+         * exposes details of the UI that are not represented on the
+         * TimeConductor API itself such as modes and deltas.
+         *
+         * @memberof platform.features.conductor
+         * @param conductor
+         * @param timeSystems
+         * @constructor
+         */
+        function TimeConductorService(openmct) {
+
+            EventEmitter.call(this);
+
+            this.timeAPI = openmct.time;
+
+        }
+
+        TimeConductorService.prototype = Object.create(EventEmitter.prototype);
+        /**
+         * @private
+         * @param {number} time some value that is valid in the current TimeSystem
+         */
+        TimeConductorService.prototype.tick = function (time) {
+            var deltas = this.deltas();
+            var startTime = time;
+            var endTime = time;
+
+            if (deltas) {
+                startTime = time - deltas.start;
+                endTime = time + deltas.end;
+            }
+
+            var newBounds = {
+                start: startTime,
+                end: endTime
+            };
+
+            this.timeAPI.boundsVal = newBounds;
+            this.timeAPI.emit('bounds', true);
+
+            // If a bounds change results in a TOI outside of the current
+            // bounds, unset it
+            if (this.timeAPI.toi < newBounds.start || this.timeAPI.toi > newBounds.end) {
+                this.timeAPI.timeOfInterest(undefined);
+            }
+        };
+
+        TimeConductorService.prototype.configuration = function (config) {
+            this.config = config;
+        };
+
+        TimeConductorService.prototype.activeTickSource = function (key) {
```

Comment:
Ah, another complication here that I had forgotten about is that having ticking on the Time API requires knowledge of deltas which are currently not a concern of the Time API, but rather the UI. There are ways around this, of inferring the deltas without explicitly managing them, but it's a bit ugly and not particularly robust. The best best might be to manage deltas on the Time API. This is not ideal as it complicates the API, but it might be the lesser of two evils.